dist/crypto.js

```diff
@@ -0,0 +1,291 @@
+      name: 'PBKDF2',
+      salt: salt || new Uint8Array([]),
+      iterations: iterations || 100000,
+      hash: hash || 'SHA-256'
+    }, baseKey, 128);
+  }).then(function (derivedKey) {
+    return new Uint8Array(derivedKey);
+  });
+};
+
+/**
+ * Hash of a string or arrayBuffer
+ *
+ * @param {string | arrayBuffer} msg The message
+ * @param {string} [type] The hash name (SHA-256 by default)
+ * @returns {Promise<Uint8Array>}   A promise that contains the hash as a Uint8Array
+ */
+var hash256 = function hash256(msg) {
+  var type = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'SHA-256';
+
+  return window.crypto.subtle.digest({
+    name: 'SHA-256'
+  }, typeof msg === 'string' ? Buffer.from(msg) : msg).then(function (digest) {
+    return new Uint8Array(digest);
+  });
+};
+
+/**
+ * Derive a passphrase and return the object to store
+ *
+ * @param {string | arrayBuffer} passPhrase The passphrase that is used to derive the key
+ * @returns {Promise<HashedPassphrase>}   A promise that contains the derived key
+ */
+var derivePassphrase = function derivePassphrase(passPhrase, salt) {
+  _checkPassphrase(passPhrase);
+  var _salt = salt || genRandomBuffer(16);
+  var iterations = 100000;
+  return deriveBitsAndHash(passPhrase, _salt, iterations).then(function (hashedValue) {
+    return {
+      salt: Buffer.from(_salt).toString('hex'),
+      iterations: iterations,
+      hashAlgo: 'SHA-256',
+      storedHash: Buffer.from(hashedValue).toString('hex')
+    };
+  }).catch(function (err) {
+    return console.err(err);
+  });
+};
+
+/**
+ * Derive the passphrase with PBKDF2 and hash the output with the given hash function
+ *
+ * @param {string | arrayBuffer} passPhrase The passphrase that is used to derive the key
+ * @param {arrayBuffer} [salt] The salt
+ * @param {Number} [iterations] The iterations number
+ * @param {string} [hash] The hash function used for derivation and final hash computing
+ * @returns {Promise<Uint8Array>}   A promise that contains the hashed derived key
+ */
+var deriveBitsAndHash = function deriveBitsAndHash(passPhrase, salt, iterations, hash) {
+  return deriveBits(passPhrase, salt, iterations, hash).then(hash256);
+};
+
+/**
+ * Check a given passphrase by comparing it to the stored hash value (in HashedPassphrase object)
+ *
+ * @param {string} passphrase The passphrase
+ * @param {HashedPassphrase} hashedPassphrase The HashedPassphrase object
+ * @returns {Promise<Boolean>}   A promise
+ */
+var checkPassphrase = function checkPassphrase(passPhrase, hashedPassphrase) {
+  _checkPassphrase(passPhrase);
+  (0, _errors.checkObject)(hashedPassphrase, requiredParameterHashedPassphrase);
+  var salt = hashedPassphrase.salt,
+      iterations = hashedPassphrase.iterations,
+      storedHash = hashedPassphrase.storedHash,
+      hashAlgo = hashedPassphrase.hashAlgo;
+
+  return deriveBitsAndHash(passPhrase, Buffer.from(salt, 'hex'), iterations, hashAlgo).then(function (hashedValue) {
```

Comment:
Can you indent all the `then` to the next line please

Reply:
Src is ok, this is the dist file.